src/core.jl

```diff
@@ -0,0 +1,170 @@
+"""
+Type representing any object drawable on image
+"""
+abstract Drawable
+
+"""
+    p = Point(x,y)
+    p = Point(c)
+
+A `Drawable` point on the image
+"""
+immutable Point <: Drawable
+    x::Int
+    y::Int
+end
+
+abstract Line <: Drawable
+abstract Circle <: Drawable
+
+
+"""
+    line = LineTwoPoints(p1, p2)
+
+A `Drawable` infinite length line passing through the two points
+`p1` and `p2`.
+"""
+immutable LineTwoPoints <: Line
+    p1::Point
+    p2::Point
+end
+
+"""
+    line = LineNormal(ρ, θ)
+
+A `Drawable` infinte length line having perpendicular length `ρ` from
+origin and angle `θ` between the perpendicular and x-axis
+"""
+immutable LineNormal <: Line
+    ρ::Real
+    θ::Real
+end
+
+"""
+    circle = CircleThreePoints(p1, p2, p3)
+
+A `Drawable` circle passing through points `p1`, `p2` and `p3`
+"""
+immutable CircleThreePoints <: Circle
+    p1::Point
+    p2::Point
+    p3::Point
+end
+
+"""
+    circle = CirclePointRadius(p, ρ)
+
+A `Drawable` circle having center `p` and radius `ρ`
+"""
+immutable CirclePointRadius <: Circle
+    center::Point
+    ρ::Real
+end
+
+"""
+    ls = LineSegment(p1, p2)
+
+A `Drawable` finite length line between `p1` and `p2`
+"""
+immutable LineSegment <: Drawable
+    p1::Point
+    p2::Point
+end
+
+"""
+    path = Path([point])
+
+A `Drawable` sequence of line segments
+"""
+immutable Path <: Drawable
+    vertices::Vector{Point}
+end
+
+"""
+    ellipse = Ellipse(center, ρx, ρy)
+
+A `Drawable` ellipse with center `center` and parameters `ρx` and `ρy`
+"""
+immutable Ellipse <: Drawable
+    center::Point
+    ρx::Real
+    ρy::Real
+end
+
+"""
+    polygon = Polygon([vertex])
+
+A `Drawable` polygon represented by a `Vector` of vertices
```

Comment:
What's the difference between a Path and a Polygon? Assuming it's closure, would be good to mention.